### ISSUE TITLE
Error in installing 'stxBrain' data from the SeuratData

### ISSUE THREAD

**Issue by vHelmholtz** [satijalab/seurat-data#19]
Hello Satijalab,

I have an error in installing 'stxBrain' data from the SeuratData. Could you please give me any advice about this issue?

**> InstallData("stxBrain")**
Installing package into ‘C:/Users/jongshin.kim/Documents/R/win-library/3.6’
(as ‘lib’ is unspecified)
Warning: dependency ‘Seurat’ is not available
trying URL 'http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz'
Content type 'application/octet-stream' length 110346050 bytes (105.2 MB)
downloaded 105.2 MB

* installing *source* package 'stxBrain.SeuratData' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
Error: package 'Seurat' required by 'stxBrain.SeuratData' could not be found
Execution halted
ERROR: lazy loading failed for package 'stxBrain.SeuratData'
* removing 'C:/Users/jongshin.kim/Documents/R/win-library/3.6/stxBrain.SeuratData'

The downloaded source packages are in
	‘C:\Users\Public\Documents\ESTsoft\CreatorTemp\RtmpgPtoMI\downloaded_packages’
Error in loadNamespace(name) : 
  there is no package called ‘stxBrain.SeuratData’
In addition: Warning message:
In install.packages(pkgs = pkgs, repos = getOption(x = "SeuratData.repo.use"),  :
  installation of package ‘stxBrain.SeuratData’ had non-zero exit status

Even though I tried the code recommended in #15, it did not work as below.

**> install.packages("http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz", repos = NULL, type = "source")**
Installing package into ‘C:/Users/jongshin.kim/Documents/R/win-library/3.6’
(as ‘lib’ is unspecified)
trying URL 'http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz'
Content type 'application/octet-stream' length 110346050 bytes (105.2 MB)
downloaded 105.2 MB

* installing *source* package 'stxBrain.SeuratData' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
Error: package 'Seurat' required by 'stxBrain.SeuratData' could not be found
Execution halted
ERROR: lazy loading failed for package 'stxBrain.SeuratData'
* removing 'C:/Users/jongshin.kim/Documents/R/win-library/3.6/stxBrain.SeuratData'
Warning in install.packages :
  installation of package ‘C:/Users/Public/Documents/ESTsoft/CreatorTemp/RtmpgPtoMI/downloaded_packages/stxBrain.SeuratData_0.1.1.tar.gz’ had non-zero exit status

**Comment by vHelmholtz** [comment]
Now, the installation of "stxBrain" is working.

**Comment by mieko-cc** [comment]
> Now, the installation of "stxBrain" is working.

could you tell me how to install this data, i suffered the same question, thank you!

**Comment by huangch** [comment]
Tried spxBrain and got the same error.

**Comment by cpusummer-wdn** [comment]
I had the same question.

**Comment by ChenJiyangOnline** [comment]
I suffered the same question as well. 
When I perform the command "InstallData("stxBrain")", it reported the followinng error message:
 
trying URL 'http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz'
Content type 'application/octet-stream' length 110346050 bytes (105.2 MB)
downloaded 32.0 MB

Error in download.file(url, destfile, method, mode = "wb", ...) : 
  download from 'http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz' failed
In addition: Warning messages:
1: In download.file(url, destfile, method, mode = "wb", ...) :
  downloaded length 0 != reported length 0
2: In download.file(url, destfile, method, mode = "wb", ...) :
  URL 'https://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz': Timeout of 60 seconds was reached
Warning in download.packages(pkgs, destdir = tmpd, available = available,  :
  download of package ‘stxBrain.SeuratData’ failed
Error in loadNamespace(name) : 
  there is no package called ‘stxBrain.SeuratData’
  
  
  Therefore, I tried the command mentioned upper: "install.packages("http://seurat.nygenome.org/src/contrib/stxBrain.SeuratData_0.1.1.tar.gz", repos = NULL, type = "source")" 
  It worked. 
  
  My R.version is  R version 4.3.0 (2023-04-21 ucrt).
  
![Uploading 微信图片_20230710094204.png…]()
